macros/src/optional.rs

```diff
@@ -77,17 +77,15 @@ pub fn apply(
         // explicit `#[ts(optional)]` on field.
         // It takes precedence over the struct attribute, and is enforced **AT COMPILE TIME**
         (_, Optional::Optional { nullable }) => (
-            // expression that evaluates to the string "?", but fails to compile if `ty` is not an `Option`.
-            parse_quote_spanned! { span => {
-                fn check_that_field_is_option<T: #crate_rename::IsOption>(_: std::marker::PhantomData<T>) {}
-                let x: std::marker::PhantomData<#field_ty> = std::marker::PhantomData;
-                check_that_field_is_option(x);
-                true
-            }},
+            parse_quote!(true),
             if nullable {
                 field_ty.clone()
             } else {
-                unwrap_option(crate_rename, field_ty)
+                // expression that evaluates to the the Option's inner type,
+                // but fails to compile if `field_ty` is not an `Option`.
+                parse_quote_spanned! {
+                    span => <#field_ty as #crate_rename::IsOption>::Inner
+                }
```

Comment:
This is simpler than the previous check, though I suspect it sacrifices on the diagnostic emitted.